mysql-test/suite/versioning/t/partition.test

```diff
@@ -550,6 +550,38 @@ update t1 set a = 4;
 delete from t1;
 delete from t1 where a is not null;
 
+--echo # MDEV-14675 Questionable semantics of INTERVAL values in PARTITION BY system_time: converted into seconds
+create or replace table t1 (i int) with system versioning partition by system_time interval 1 year (partition p0 history, partition pn current);
+--replace_result $default_engine DEFAULT_ENGINE
+show create table t1;
+create or replace table t1 (i int) with system versioning partition by system_time interval 1 quarter (partition p0 history, partition pn current);
+--replace_result $default_engine DEFAULT_ENGINE
+show create table t1;
+create or replace table t1 (i int) with system versioning partition by system_time interval 1 month (partition p0 history, partition pn current);
+--replace_result $default_engine DEFAULT_ENGINE
+show create table t1;
+create or replace table t1 (i int) with system versioning partition by system_time interval 1 week (partition p0 history, partition pn current);
+--replace_result $default_engine DEFAULT_ENGINE
+show create table t1;
+create or replace table t1 (i int) with system versioning partition by system_time interval 1 day (partition p0 history, partition pn current);
+--replace_result $default_engine DEFAULT_ENGINE
+show create table t1;
+create or replace table t1 (i int) with system versioning partition by system_time interval 1 hour (partition p0 history, partition pn current);
+--replace_result $default_engine DEFAULT_ENGINE
+show create table t1;
+create or replace table t1 (i int) with system versioning partition by system_time interval 1 minute (partition p0 history, partition pn current);
+--replace_result $default_engine DEFAULT_ENGINE
+show create table t1;
+create or replace table t1 (i int) with system versioning partition by system_time interval 1 second (partition p0 history, partition pn current);
+--replace_result $default_engine DEFAULT_ENGINE
+show create table t1;
```

Comment:
5. Remove all the above except `year` test. Don't test parser, this is not your code. In future restrain from adding redundant cases.

Reply:
6. Test rotation by `year`. You may set arbitrary dates for NOW time. See examples in this file.

Reply:
I can't agree with `5.`.

For example, I've applied this patch and see and assertion failure. This code wan't tested:
```diff
diff --git a/sql/partition_info.cc b/sql/partition_info.cc
index c258d3e74e1..367a70b8794 100644
--- a/sql/partition_info.cc
+++ b/sql/partition_info.cc
@@ -874,7 +874,7 @@ bool partition_info::vers_init_info(THD * thd)
 
 bool partition_info::vers_set_interval(const INTERVAL & i)
 {
-  if (i.neg || i.second_part || i.empty())
+  if (i.second_part || i.empty())
     return true;
 
   DBUG_ASSERT(vers_info);
```

Another one. Now parser error:
```diff
diff --git a/sql/sql_partition.cc b/sql/sql_partition.cc
index e3652c89d2a..fcac365c409 100644
--- a/sql/sql_partition.cc
+++ b/sql/sql_partition.cc
@@ -2346,7 +2346,6 @@ char *generate_partition_syntax(THD *thd, partition_info *part_info,
         err+= str.append_ulonglong(interval.year);
         break;
       case INTERVAL_MONTH:
-        err+= str.append_ulonglong(interval.month);
         break;
       case INTERVAL_DAY:
         err+= str.append_ulonglong(interval.day);
```

So, these tests are needed.